### ISSUE TITLE
redundant do with when and when-not is not reported

### ISSUE THREAD

**Issue by borkdude** [borkdude/clj-kondo#710]
```
$ clj-kondo --lint - <<< '(when true (do 1 2 3))'
linting took 9ms, errors: 0, warnings: 0
```
